### PR TITLE
add store finished message on Tls13

### PR DESCRIPTION
**Pull request by miyazakh**

### CODE REVIEW

File: src/tls13.c [hunk 1]
```diff
@@ -6870,13 +6870,15 @@ int DoTls13Finished(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
...
-        }
-        else {
-            XMEMCPY(ssl->clientFinished, mac, finishedSz);
-            ssl->clientFinished_len = finishedSz;
+            if (finishedSz <= TLS_FINISHED_SZ_MAX) {
+                XMEMCPY(ssl->serverFinished, mac, finishedSz);
+                ssl->serverFinished_len = finishedSz;
+            }
         }
+        else if (finishedSz <= TLS_FINISHED_SZ_MAX) {
```
Comment (SparkiDev) [inline src/tls13.c:6878]: Prefer check of finishSz around outside check of side.
Why would this happen? We are building the MAC and we should know the maximum size.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/tls13.c (hunk 1 above)
Comment (miyazakh) [reply src/tls13.c:6878]: Agree. This is redundant. Removed the check

File: src/ssl.c [hunk 2]
```diff
@@ -28310,37 +28310,43 @@ int wolfSSL_i2d_ASN1_OBJECT(WOLFSSL_ASN1_OBJECT *a, unsigned char **pp)
...
+        XMEMCPY(buf, ssl->clientFinished, len);
+    }
+    return len;
 }
 
 WOLFSSL_API size_t wolfSSL_get_peer_finished(const WOLFSSL *ssl, void *buf, size_t count)
 {
     WOLFSSL_ENTER("SSL_get_peer_finished");
-
+    byte len = 0;
```
Comment (SparkiDev) [inline src/ssl.c:28334]: Move this declaration above the potential code line.
Older compilers don't allow this.

File: src/tls13.c [hunk 3]
```diff
@@ -6868,6 +6868,16 @@ int DoTls13Finished(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
 
     if (sniff == NO_SNIFF) {
         ret = BuildTls13HandshakeHmac(ssl, secret, mac, &finishedSz);
+    #if defined(OPENSSL_ALL) || defined(WOLFSSL_HAPROXY) || defined(WOLFSSL_WPAS)
+        if (ssl->options.side == WOLFSSL_CLIENT_END) {
+                XMEMCPY(ssl->serverFinished, mac, finishedSz);
```
Comment (SparkiDev) [inline src/tls13.c:6873]: Bring in one indent.

**Review by SparkiDev — CHANGES_REQUESTED** [review]
Formatting fixes, otherwise looks fine!

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**